specviz/third_party/glue/data_viewer.py

```diff
@@ -0,0 +1,208 @@
+
+        self.setCentralWidget(self.viewer.main_window)
+
+    # The following method is required by glue - it is used to subscribe the
+    # viewer to various messages sent by glue.
+
+    def register_to_hub(self, hub):
+        super(SpecVizViewer, self).register_to_hub(hub)
+
+        hub.subscribe(self, msg.SubsetCreateMessage,
+                      handler=self._add_subset)
+
+        hub.subscribe(self, msg.SubsetUpdateMessage,
+                      handler=self._update_subset)
+
+        hub.subscribe(self, msg.SubsetDeleteMessage,
+                      handler=self._remove_subset)
+
+        hub.subscribe(self, msg.DataUpdateMessage,
+                      handler=self._update_data)
+
+    def _spectrum_from_component(self, component, wcs, mask=None):
+        data = SpectralCube(component.data, wcs)
+
+        # if mask is not None:
+        #     data = data.with_mask(~mask)
+
+        spec_data = data.sum((1, 2))
+
+        spec_data = Spectrum1DRef(spec_data.data,
+                                  unit=spec_data.unit,
+                                  dispersion=data.spectral_axis.data,
+                                  dispersion_unit=data.spectral_axis.unit,
+                                  wcs=data.wcs)
+
+        dispatch.on_add_to_window.emit(spec_data)
+
+    def _update_combo_boxes(self, data):
+        if data not in self._layer_widget:
+            self._layer_widget.add_layer(data)
+
+        self._layer_widget.layer = data
+        self._options_widget.set_data(self._layer_widget.layer)
+
+        if self._options_widget.file_att is None:
+            return False
+
+        if self._layer_widget.layer is None:
+            return False
+
+        return True
+
+    # The following two methods are required by glue - they are what gets called
+    # when a dataset or subset gets dragged and dropped onto the viewer.
+
+    def add_data(self, data):
+        if not self._update_combo_boxes(data):
+            return
+
+        cid = self._layer_widget.layer.id[self._options_widget.file_att]
+        mask = None
+        component = self._layer_widget.layer.get_component(cid)
+
+        self._spectrum_from_component(component, data.coords.wcs)
+
+        return True
+
+    def add_subset(self, subset):
+        if not self._update_combo_boxes(subset):
+            return
+
+        subset = self._layer_widget.layer
+        cid = subset.data.id[self._options_widget.file_att]
+        mask = subset.to_mask()
+        component = subset.data.get_component(cid)
+
+        self._spectrum_from_component(component, subset.data.coords.wcs, mask=mask)
+
+        return True
+
+    # The following four methods are used to receive various messages related
+    # to updates to data or subsets.
+
+    def _update_data(self, message):
+        print("Updating data")
+
+    def _add_subset(self, message):
+        self.add_subset(message.subset)
+
+    def _update_subset(self, message):
+        print("Updating subset")
```

Comment:
``_update_data``, ``_update_subset``, and ``_remove_subset`` should probably be implemented before merging?